vxfreeswitch/tests/test_voice.py

```diff
@@ -444,3 +462,102 @@ def test_reply_to_client_that_has_hung_up(self):
         self.assertEqual(nack['user_message_id'], msg['message_id'])
         self.assertEqual(nack['nack_reason'],
                          "Client u'TESTTEST' no longer connected")
+
+
+class RecordingServer(Protocol):
+    def connectionMade(self):
+        self.factory.clients.append(self)
+
+    def _send_event(self, content):
+        self.transport.write(
+            'Content-Length: %s\n' % len(content) +
+            'Content-Type: text/event-plain\n\n' +
+            content)
+
+    def dataReceived(self, line):
+        self.factory.data.append(line)
+        uuid = '%s' % uuid4()
+        # Send server immediate response
+        self.transport.write(
+            'Content-Type: command/reply\n'
+            'Reply-Text: +OK\n'
+            'Job-UUID: %s\n\n' % uuid)
+        # Send job complete success response
+        if not line.startswith('bgapi originate'):
+            return
+        content = (
+            'Content-Length: %s\n' % (len(uuid) + 4) +
+            'Event-Name: BACKGROUND_JOB\n' +
+            'Job-UUID: %s\n\n' % uuid +
+            '+OK %s\n' % uuid)
+        self._send_event(content)
+
+    def hangup(self):
+        content = (
+            'Event-Name: CHANNEL_HANGUP\n')
+        self._send_event(content)
+
+
+class RecordingServerFactory(protocol.Factory):
+    protocol = RecordingServer
+    data = []
+    clients = []
```

Comment:
We should set ``data`` and ``clients`` in ``__init__`` since they're mutable (and thus clients might hang around between tests if we don't).